对项目宏ENABLE_WEBGL_HIGHP_STRUCT_VALUES的说明文档 (#1943)
cocos-creator/creator-docs: en/editor/project/index.md

- **MAX_LABEL_CANVAS_POOL_SIZE**: set the maximum number of Canvas object pools used by the Label, and adjust it according to the number of Labels in the same scene of the project.

> - **ENABLE_WEBGL_HIGHP_STRUCT_VALUES**：whether to enable WebGL highp precision with fragment shader struct variables. These variables are mediump precision with Android+WebGL platform. It may lead to light calculation errors of some data requiring highp precision (such as position)。

Reviewer: ```suggestion
- **ENABLE_WEBGL_HIGHP_STRUCT_VALUES**: on the Android platform with WebGL backend, the internal variables of the structures defined in the fragment shader use **mediump** precision, which may lead to incorrect results for some data that requires high precision (such as position information). To avoid this problem, check this option (new in v3.4.1) and enable WebGL to calculate variables with **highp** precision.

```

>     - If enabled, you should use HIGHP_VALUE_STRUCT_DEFINE macro to define variables in fragment shader struct which needs to be highp precision, use HIGHP_VALUE_TO_STRUCT_DEFINED and HIGHP_VALUE_FROM_STRUCT_DEFINED macro to do assignment between struct varibles and non-struct variables.

Reviewer: ```suggestion
    - If this option is enabled, use the `HIGHP_VALUE_STRUCT_DEFINE` macro in the fragment shader code to define structural variables, and use `HIGHP_VALUE_TO_STRUCT_DEFINED` and `HIGHP_VALUE_FROM_STRUCT_DEFINED` to assign values between structural and non-structural variables.
```

>     - For more specific information and code about these macros, please refer to the [packing.chunk](https://github.com/cocos-creator/engine/blob/develop/editor/assets/chunks/packing.chunk#L40).

Reviewer: ```suggestion
    - For specific information and code on the above macro calls, please refer to [packing.chunk](https://github.com/cocos-creator/engine/blob/v3.4.0/editor/assets/chunks/packing.chunk] for details.
```

- **MAX_LABEL_CANVAS_POOL_SIZE**: set the maximum number of Canvas object pools used by the Label, and adjust it according to the number of Labels in the same scene of the project.

- **ENABLE_WEBGL_HIGHP_STRUCT_VALUES**: on the Android platform with WebGL backend, the internal variables of the structures defined in the fragment shader use **mediump** precision, which may lead to incorrect results for some data that requires high precision (such as position information). To avoid this problem, check this option (new in v3.4.1) and enable WebGL to calculate variables with **highp** precision.

    - If this option is enabled, use the `HIGHP_VALUE_STRUCT_DEFINE` macro in the fragment shader code to define structural variables, and use `HIGHP_VALUE_TO_STRUCT_DEFINED` and `HIGHP_VALUE_FROM_STRUCT_DEFINED` to assign values between structural and non-structural variables.

>     - For specific information and code on the above macro calls, please refer to [packing.chunk](https://github.com/cocos-creator/engine/blob/v3.4.1/editor/assets/chunks/packing.chunk#L40] for details.

Reviewer: ```suggestion
    - For specific information and code on the above macro calls, please refer to [packing.chunk](https://github.com/cocos-creator/engine/blob/v3.4.1/editor/assets/chunks/packing.chunk#L40) for details.
```
